huggingface_pipelines/builder.py

```diff
@@ -0,0 +1,129 @@
+from .pipeline import Pipeline, PipelineFactory  # type: ignore
+from .text import (  # type: ignore[import]
+    EmbeddingToTextPipelineFactory,
+    TextSegmentationPipelineFactory,
+    TextToEmbeddingPipelineFactory,
+)
+
+logger = logging.getLogger(__name__)
+
+# Define a custom type for supported operations
+SupportedOperation = Literal[
+    "text_to_embedding",
+    "embedding_to_text",
+    "text_segmentation",
+    "analyze_metric",
+    "audio_to_embedding",
+]
+
+
+class PipelineBuilder:
+    """
+    A class for building and managing different types of processing pipelines.
+
+    This class provides methods to create pipelines for various operations such as
+    text-to-embedding, embedding-to-text, text segmentation, metric analysis, and
+    audio-to-embedding. It uses the Factory pattern to create specific pipeline
+    instances based on the operation type and configuration.
+
+
+    Attributes:
+        config_dir (Path): The directory containing configuration files for pipelines.
+        pipeline_factories (Dict[SupportedOperation, PipelineFactory]): A dictionary mapping
+            operations to their respective factory classes.
+
+    Args:
+        config_dir (Union[str, Path], optional): The directory containing configuration
+            files. Defaults to "huggingface_pipelines/datacards".
+
+    Example:
+        >>> builder = PipelineBuilder()
+        >>> text_to_embedding_pipeline = builder.create_pipeline("dataset_name", "text_to_embedding")
+        >>> processed_dataset = text_to_embedding_pipeline(input_dataset)
+    """
+
+    def __init__(
+        self, config_dir: Union[str, Path] = "huggingface_pipelines/datacards"
+    ):
+        self.config_dir = Path(config_dir)
+        self.pipeline_factories: Dict[SupportedOperation, PipelineFactory] = {
+            "text_to_embedding": TextToEmbeddingPipelineFactory(),
+            "embedding_to_text": EmbeddingToTextPipelineFactory(),
+            "text_segmentation": TextSegmentationPipelineFactory(),
+            "analyze_metric": MetricAnalyzerPipelineFactory(),
+            "audio_to_embedding": AudioToEmbeddingPipelineFactory(),
+        }
+
+    def load_config(
+        self, dataset_name: str, operation: SupportedOperation
+    ) -> Dict[str, Any]:
+        """
+        Load the configuration for a specific dataset and operation.
+
+        This method reads the YAML configuration file for the specified dataset and operation.
+        The configuration is used to set up the PipelineConfig for the requested pipeline.
+
+        Args:
+            dataset_name (str): The name of the dataset.
+            operation (SupportedOperation): The type of operation to perform.
+
+        Returns:
+            Dict[str, Any]: The configuration dictionary.
+
+        Raises:
+            FileNotFoundError: If the configuration file is not found.
+
+        """
+        config_file = self.config_dir / f"{dataset_name}/{operation}.yaml"
```

Comment:
can you add an example of such cards ?
btw, please look how it's done in sonar with model cards (where all loading logic is already done)